### ISSUE TITLE
提个建议

### ISSUE THREAD

**Issue by wuerpeng** [terrylinooo/cache-master#37]
是只能用户访问的时候才可以触发生成缓存,且生命周期最多只能设置1天吗？
能不能全部文章都生成一次,到生命周期的时候用户访问再生成缓存给下一个用户访问？
这样做的好处就是所有文章缓存起来后，大家都访问缓存，也就不会触发500 499 502状态了，也就是说不管数据库还是php是不是挂了，一律都是优先访问缓存，尽可能最大化返回200状态给拥护和蜘蛛
生命周期可以自己设置
例如
首页 1天
分类 7天
文章 永久
首页比如现在到期了，有访客去访问一下，这时候还是老缓存，但这时后台因生命周期到期且有新访问会悄悄生成新缓存，下次别的用户的访问时候就是新缓存了

**Comment by terrylinooo** [comment]
謝謝你的建議，生命週期我再擴增功能喔，把它獨立一個區塊設定。謝謝你的建議。
使用上如有問題歡迎和我說

**Comment by wuerpeng** [comment]
没想到回复那么快！
这个需求原因在于，我们使用缓存就是为了避免高消耗php性能时产生502状态或者其它出错状态，以及提高用户访问站点的加载速度，毕竟wordpress所有的文章都存放在一个表中，超出10万文章不仅前台包括后台都会慢。
因此我们需要一个强大的缓存系统的存在。
经过朋友推荐，发现了您这款非常强大且符合预期的支持sqlite缓存插件，只不过目前还不支持预缓存、发布文章自动生成缓存和生命周期这些功能。

请允许我复述一下为什么需要这些功能
搜索引擎对于站点的速度也是重要的考核之一，目前的缓存生成是靠用户访问才会生成，如果用户访问的时候正好服务器超出负荷返回给502状态，那就对用户不太友好了，包括蜘蛛访问返回的状态是502的话，很容易触发降权的风险；所以我希望您可以支持预缓存，将所有的文章都生成缓存，且全部文章缓存生成之后，每发布一篇文章就产生当前文章的缓存，这样不仅仅搜索引擎蜘蛛也好或许用户访问也好，都会直接访问缓存，如果伪静态（动态）访问速度是在1秒-3秒之间，那么访问缓存会在0.3秒左右，这对于非交互站点非常实用，也对于搜索引擎的收录和排名更加友好。

生命周期，伪静态（动态）是没有立即获取最新的内容变化，但是有缓存情况下修改文章时自动更新缓存也是有必要的，自动声明周期的使用场景是：文章页侧边栏的一些模块例如最新文章是每天都会有变化，文章页设置生命周期为1天，第二天用户访问的时候自动产生最新的缓存让下一个用户访问新的缓存。

**Comment by terrylinooo** [comment]
了解了。我這個星期再更新這個功能。

**Comment by wuerpeng** [comment]
> 

感谢！如果有需要我协助的地方，我义不容辞

**Comment by wuerpeng** [comment]
我想请教您一下

1:缓存选择文件模式,这文件默认存放在什么位置?

2:缓存选择sqlite模式,如果缓存超出10W以上文章数据,sqlite会不会自动进行分库?从而避免全部文章的缓存都放在一个sqlite数据库文件里

**Comment by wuerpeng** [comment]
发现个细节问题
多站点模式下,主站点首页生成了缓存之后,其它子站点的首页都是主站点的首页了

**Comment by terrylinooo** [comment]
> 我想请教您一下
> 
> 1:缓存选择文件模式,这文件默认存放在什么位置?

wp-content\uploads\cache-master

> 
> 2:缓存选择sqlite模式,如果缓存超出10W以上文章数据,sqlite会不会自动进行分库?从而避免全部文章的缓存都放在一个sqlite数据库文件里

不會將資料分表分庫喔。如果您的資料有 10 萬筆，假設每張網頁 300 KB，需要的空間高達 30GB 以上。
這個外掛是基於我寫的 PHP Simple Cache 函式庫
https://github.com/terrylinooo/simple-cache
都是幾 Cache key 為主鍵，會使用 index，不用擔心效率問題。不過 30 GB 以上的 SQLite 似乎不適用 SQL "Lite" 的使用場景，我會比較建議您另外起一台小 VPS 掛 MongoDB 比較適合喔。
不過如果您的空間夠的話，使用 SQLite 還是可以的。

我今晚會更新一個讓 Cache 不過期的選項讓您使用

![image](https://github.com/terrylinooo/cache-master/assets/11989371/3df6d559-2bfe-47f2-90c1-ad5e744f6312)

至於 10 萬筆文章的情境，自產 Cache 的功能，我有空時再補上

**Comment by wuerpeng** [comment]
首先谢谢您，做出这么好的产品和可以听取我们的小建议，且及时更新造福我们使用场景。

我想额外再请教您，因为现在不清楚究竟哪个缓存适合了:

站点固定链接并非以.html扩展名为链接
而是以 "站点域名/分类/文章id/" 作为的固定链接
所以首先就pass掉文件名的方式,否则生成的都是文件夹，也不太好管理；

redis和memcached，存放在内存中，我服务器128G内存这个是够用，但因为文章发布的时候有图片本地化的功能，用redis的话，文章发布之后redis缓存显示的是图片没有本地化时的数据，且暂时还无解，所以pass掉了；

sqlite作为缓存驱动，这个接触过，且服务器硬盘也足够，我提议分库的原因不是因为硬盘，而是负载上的细节建议，几十万数据都在一个sqlite数据库（如果多站点也都是在一个sqlite数据库中），频繁查询这一个库，担心会造成查询较慢的情况。

MongoDB和其它的缓存驱动还没有体验过...

站点较多，且都是用wordpress多站点模式，数据也比较多

您是专业的，可以不可以指点迷津？这种情况哪一种缓存驱动是性能卓越的？

**Comment by terrylinooo** [comment]
多站點的情況下，不同的站點它的 SQLite 檔案是不同的喔。

**Comment by wuerpeng** [comment]
> 多站點的情況下，不同的站點它的 SQLite 檔案是不同的喔。

是的,就是类似于wordpress每个站点所有的文章都是放在wp(站点id)_posts这个表中,文章越多这个表就越大,表越大查询速度就越慢,我担心sqlite也会类似于这样

**Comment by wuerpeng** [comment]
请问开发者近期有更新的计划嘛?
